default.nix

```diff
@@ -604,9 +605,9 @@ let overrideCabal = pkg: f: if pkg == null then null else lib.overrideCabal pkg
 in let this = rec {
   overrideForGhcjs = haskellPackages: haskellPackages.override {
     overrides = self: super: {
-      mkDerivation = drv: super.mkDerivation.override {
-        hscolour = ghc.hscolour;
-      } (drv // {
+      hscolour = ghc.hscolour;
```

Comment:
Well, my understanding (which may be wrong) of the previous way was that any mentions of hscolour explicitly in mkDerivation would use the one we passed in with `.override`, but package lookups would still use `self`.  I think the key is to make sure that we can compile ghcjs-based packages that depend on hscolour as well as using haddock to produce hscoloured documentation for ghcjs-based packages.  If we can do both, then the details of how it works aren't quite as important.

It occurs to me that whatever refactor took `.override` away could possibly have fixed the issue permanently, so that we don't even need to worry about this at all; however, we should probably add a test case to `test.hs`.